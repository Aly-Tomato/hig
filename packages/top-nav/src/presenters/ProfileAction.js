import React from "react";
import PropTypes from "prop-types";
import Avatar from "@hig/avatar";
import { combineEventHandlers } from "@hig/utils";
/** @todo Move flyout into its own package */
import Flyout from "hig-react/lib/elements/components/Flyout";

import "@hig/avatar/build/index.css";

import Action from "./Action";

export default function ProfileAction({
  avatarName,
  avatarImage,
  onClick,
  children
}) {
  return (
    <Action>
      <Flyout anchorPoint="top-right" content={children}>
        {({ handleClick }) => (
          <button
            type="button"
            className="hig__top-nav__profile-action"
            onClick={combineEventHandlers(onClick, handleClick)}
          >
            <Avatar name={avatarName} image={avatarImage} size="medium-32" />
          </button>
        )}
      </Flyout>
    </Action>
  );
}

ProfileAction.propTypes = {
  /** The name that will converted into initials, and displayed when an image isn't available */
  avatarName: PropTypes.string.isRequired,
  /** Url to a profile image */
  avatarImage: PropTypes.string,
  /** Content to render in the profile flyout */
  children: PropTypes.node,
  /** Callback when the flyout is opened */
  onClick: PropTypes.func
};
