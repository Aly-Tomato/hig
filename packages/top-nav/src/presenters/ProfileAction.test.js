import React from "react";
import renderer from "react-test-renderer";

import ProfileAction from "./ProfileAction";

describe("top-nav/ProfileAction", () => {
  const cases = [
    {
      description: "renders with avatarName",
      props: {
        avatarName: "Wolfgang Pauli"
      }
    },
    {
      description: "renders with children and avatarName",
      props: {
        avatarName: "Hans Hellmann",
        children: <div className="logo" />
      }
    },
    {
      description: "renders with all props",
      props: {
        avatarName: "Richard Feynman",
        avatarImage: "richard_phillips_feynman.png",
        children: <div className="logo" />,
        onClick: () => {}
      }
    }
  ];

  cases.forEach(({ description, props }) => {
    it(description, () => {
      const tree = renderer.create(<ProfileAction {...props} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
