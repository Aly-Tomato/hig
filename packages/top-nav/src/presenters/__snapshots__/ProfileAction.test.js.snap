// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-nav/ProfileAction renders with all props 1`] = `
<div
  className="hig__top-nav__action"
>
  <div
    data-mock="flyout"
  >
    <div
      data-mock="flyout-content"
    >
      <div
        className="logo"
      />
    </div>
    <div
      data-mock="flyout-children"
    />
  </div>
</div>
`;

exports[`top-nav/ProfileAction renders with avatarName 1`] = `
<div
  className="hig__top-nav__action"
>
  <div
    data-mock="flyout"
  >
    <div
      data-mock="flyout-content"
    />
    <div
      data-mock="flyout-children"
    />
  </div>
</div>
`;

exports[`top-nav/ProfileAction renders with children and avatarName 1`] = `
<div
  className="hig__top-nav__action"
>
  <div
    data-mock="flyout"
  >
    <div
      data-mock="flyout-content"
    >
      <div
        className="logo"
      />
    </div>
    <div
      data-mock="flyout-children"
    />
  </div>
</div>
`;
